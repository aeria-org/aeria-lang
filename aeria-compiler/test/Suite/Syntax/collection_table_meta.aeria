collection User {
  properties {
    first_name  str
    last_name   str
  }

  getters {
    full_name @js (doc) => {
      return `${doc.first_name} ${doc.last_name}`
    }
    @end
  }

  tableMeta {
    full_name
  }
}
