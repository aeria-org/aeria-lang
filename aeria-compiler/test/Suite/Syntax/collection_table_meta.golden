{
  "kind": "Program",
  "collections": [
    {
      "tableMeta": [
        {
          "propertyName": {
            "value": "prop1",
            "kind": "PropertyName"
          },
          "kind": "TableMetaItem"
        },
        {
          "propertyName": {
            "value": "prop2",
            "kind": "PropertyName"
          },
          "kind": "TableMetaItem"
        },
        {
          "propertyName": {
            "value": "prop3",
            "kind": "PropertyName"
          },
          "kind": "TableMetaItem"
        }
      ],
      "name": {
        "value": "CollectionTableMetaTest",
        "kind": "CollectionName"
      },
      "kind": "Collection"
    }
  ]
}
